app/src/main/java/edu/grinnell/grinnell_publications_android/Services/Implementation/PublicationsRemoteClient.java

```diff
@@ -19,68 +30,106 @@
+      @Override public void onResponse(Call<List<JsonPublication>> call,
+                                       Response<List<JsonPublication>> response) {
+        storeRealmPublication(response.body());
+      }
+      public void onFailure(Call<List<JsonPublication>> call, Throwable t) {
+        //TODO: Throw Error
+      }
+    });
+  }
 
-    @Override
-    public Publication getPublication(int id) {
-        return null;
+  private void storeRealmPublication(List<JsonPublication> list) {
+    instantiateRealmPObject();
+    for (JsonPublication item : list) {
+      RealmPublication pub = new RealmPublication(item.getName(), item.getId(), null, null, null);
+      mRealm.copyToRealm(pub);
+      mRealm.commitTransaction();
     }
+  }
 
-    @Override
-    public List<Publication> getPublications(Set<Integer> publicationIds) {
-        return null;
-    }
+  public void getStory(String publicationId, String articleId) {
+    Call<JsonStory> call = mPubAPI.article(publicationId, articleId);
+    call.enqueue(new Callback<JsonStory>() {
+      @Override public void onResponse(Call<JsonStory> call, Response<JsonStory> response) {
+        storeRealmStory(convertToRealmStory(response.body()));
+      }
+      public void onFailure(Call<JsonStory> call, Throwable t) {
+        //TODO: Throw Error
+      }
+    });
+  }
 
-    /* Public methods */
-
-
-    //TODO : Add logging to the HTTP calls.
-    // TODO : DRY out service calls.? Repetition
-    /* Private classes  and interfaces */
-    private interface PublicationsServiceAPI {
-        /* Publications*/
-        @GET("publications/{id}")
-        Call<RemoteQueryResponse> publication( @Path("id")int publicationId);
-        /* Articles */
-        @GET("articles/{id}")
-        Call<RemoteQueryResponse> article( @Path("id")int articleId);
-        /* Images */
-        @GET("images/{id}")
-        Call<RemoteQueryResponse> image( @Path("id")int imageId);
-        /* Series */
-        @GET("series/{id}")
-        Call<RemoteQueryResponse> series( @Path("id")int seriesId);
-    }
-    /* Private helper methods */
-    private Retrofit initializeRetrofitRestAdapter() {
-        Gson gson = new GsonBuilder().create();
-        return new Retrofit.Builder()
-                .baseUrl(BASE_API).addConverterFactory(GsonConverterFactory.create(gson)).build();
+  private RealmStory convertToRealmStory(JsonStory s) {
+    RealmList realmAuthorList = new RealmList<RealmAuthor>();
+    for (JsonAuthor author : s.getAuthors()) {
+      RealmAuthorContact contact = new RealmAuthorContact(author.getEmail(), null, null);
+      realmAuthorList.add(new RealmAuthor(author.getName(), contact, null, null));
     }
+    return new RealmStory(s.getDatePublished(), s.getBrief(), s.getHeaderImage(), s.getPublication(),
+            s.getDateEdited(), s.getId(), s.getTitle(), s.getContent(), realmAuthorList);
+  }
+
+  private void storeRealmStory(RealmStory realmStory) {
```

Comment:
What happens if the `realmStory` is null? Null-checks will be good right about now